src/widgets/merch-catalog/ui/merch-catalog.tsx

```diff
@@ -1,34 +1,61 @@
+'use client';
+
+import { useMemo } from 'react';
 import classNames from 'classnames/bind';
-import Image from 'next/image';
+import { ReadonlyURLSearchParams, useSearchParams } from 'next/navigation';
 
+import { getTags } from '../helpers/get-tags';
+import { MerchProductsProps } from '../types';
+import { MerchFilter } from './merch-filter/merch-filter';
 import { MerchList } from './merch-list/merch-list';
-import { merchStore } from '@/entities/merch';
-import notFoundImg from '@/shared/assets/404.webp';
-import { Paragraph } from '@/shared/ui/paragraph';
+import { NoMerch } from './no-merch/no-merch';
+import { MerchProduct } from '@/entities/merch/types';
 
 import styles from './merch-catalog.module.scss';
 
 const cx = classNames.bind(styles);
 
-export const MerchCatalog = async () => {
-  const products = await merchStore.loadMerchCatalog();
+export const MerchCatalog = ({ products }: MerchProductsProps) => {
+  const searchParams: ReadonlyURLSearchParams = useSearchParams();
+
+  const searchTerm: string = (searchParams.get('search') || '').trim().toLowerCase();
+  const selectedTypes: string[] = searchParams.getAll('type');
+  const isFiltered: boolean = !!(searchTerm || selectedTypes.length);
+  const tags: string[] | [] = getTags(products);
+
+  const filteredProducts: MerchProduct[] | [] = useMemo(() => {
+    return products
+      .filter((product) => {
+        if (!searchTerm) {
+          return true;
+        }
+        const titleMatch = product.title.toLowerCase().includes(searchTerm);
+        const tagsMatch = (product.tags || []).some((tag) =>
+          tag.toLowerCase().includes(searchTerm),
+        );
+
+        return titleMatch || tagsMatch;
+      })
+      .filter((product) => {
+        if (selectedTypes.length === 0) {
+          return true;
+        }
+        return selectedTypes.some((type) => (product.tags || []).includes(type));
+      });
+  }, [products, searchTerm, selectedTypes]);
 
   return (
-    <section className={cx('container')}>
-      {(!products || products.length === 0) && (
-        <div className={cx('content', 'merch-catalog', 'empty-wrapper')}>
-          <Image className={cx('empty-image')} src={notFoundImg} alt="No merchandise available" />
-          <Paragraph className={cx('empty-paragraph')}>
-            No merchandise available at this time.
-          </Paragraph>
-        </div>
-      )}
-
-      {products && products.length > 0 && (
-        <div className={cx('content', 'merch-catalog')}>
-          <MerchList products={products} />
-        </div>
-      )}
-    </section>
+    <div className={cx('content', 'merch-catalog')}>
+      <div className={cx('merch-catalog-tags')}>
+        <MerchFilter tags={tags} />
+      </div>
+      {filteredProducts.length
+        ? (
+            <MerchList products={filteredProducts} />
+          )
+        : (
+            <NoMerch isFiltered={isFiltered} />
```

Comment:
Could conditions `filteredProducts.length === 0` and `isFiltered === false` be met simultaneously?